### DISCUSSION TITLE
✅ Add resource information in log drain ingest

### DISCUSSION THREAD

**Discussion by Vahor** [coollabsio/coolify#2770] [Feature Requests]
Currently, the JSON sent to the log service has this shape: (at least in axiom)
```json
{
  "container_id": "id",
  "container_name": "/something-uid",
  "log": "text",
  "server_name": "xyz",
  "source": "stdout"
}
```

Which is not really useful. It's not possible to know which service is used (the container name is not stable and not configurable). 
The only way is to achieve this today is to send a JSON message containing all the necessary information (using env vars) and parse it.

I think it would be better to have something like this
```diff
{
  "container_id": "id",
  "container_name": "/something-uid",
  "log": "text",
  "server_name": "xyz",
  "source": "stdout",

+  "environment_name": "production",
+  "project_name": "project",
+  "resource_name": "name",
}
```

name and json shape can be different. But theses informations are necessary to make proper dashboard and alerts.

There might be an interest for other fields like: tags, git branch and docker tag

Example use case: 
- alerts when there is an error in prod. (ignore other envs and preview branchs)
- graphs of requests per service

**Comment by Vahor** [comment]
Made a PR to add these new fields: https://github.com/coollabsio/coolify/pull/3237

**Reply by tribal2** [reply]
Hi, thanks for the PR that added support for passing `COOLIFY_*` env vars via the log drain 🙏

I’m trying to get these values exposed in Fluent Bit, but I don’t see them in the log records.

When I inspect a container with log drain enabled, I can see the expected env config in `LogConfig`:

```json
[
    {
        "Id": "a9706cd4f46946b38edb19a2a7eebb1b7d04d07a6baed7010dec4ce7902d02d2",
        ...
        "HostConfig": {
            "LogConfig": {
                "Type": "fluentd",
                "Config": {
                    "env": "COOLIFY_APP_NAME,COOLIFY_PROJECT_NAME,COOLIFY_SERVER_IP,COOLIFY_ENVIRONMENT_NAME",
                    "fluentd-address": "tcp://127.0.0.1:24224",
                    "fluentd-async": "true",
                    "fluentd-sub-second-precision": "true"
                }
            },
            ...
```

So Docker should be forwarding those env vars to the log driver.

But the actual output I receive from `coolify-log-drain` looks like this (no `env` block present):

```json
{
  "date": 1759426320.724337,
  "container_id": "a9706cd4f46946b38edb19a2a7eebb1b7d04d07a6baed7010dec4ce7902d02d2",
  "container_name": "/gssc4sc8488c8ksks0scoscw-170751010627",
  "source": "stdout",
  "log": "  errorMessage: null,"
}
```

Here’s the Fluent Bit config I’m using on the server:

```ini
[SERVICE]
    Flush             5
    Log_Level         debug

[INPUT]
    Name              forward
    Listen             0.0.0.0
    Port               24224
    Tag                docker

[OUTPUT]
    Name              stdout
    Match             *
    Format            json
```

I was expecting to see fields like `env.COOLIFY_APP_NAME` or similar in the log records, but they’re not there.

**Questions:**

* Should the COOLIFY_* env vars appear automatically in the log stream, or do I need extra configuration in Fluent Bit?
* Is `coolify-log-drain` dropping them somewhere?
* Do you have an example of a working config that shows how those env vars are supposed to surface?

Thanks a lot for clarifying 🙏

**Reply by djsisson** [reply]
docker passes them as metadata, they would not be included by default, you would need to setup some kind of filter to transform or modify the output

https://docs.fluentd.org/filter/record_transformer

i wouldn't know how to add this to your config, but i assume ai can assist you
